### PR TITLE
DRBG: weaken the restriction on allowed digests.

### PR DESCRIPTION
**Pull request by paulidale**
The artificial restriction of digests for the HMAC and HASH DRBGs is lifted.
Any fetchable digest is acceptable except XOF ones (such as shake).

In FIPS mode, the fetch remains internal so only a FIPS validated digest will
be located.

### CODE REVIEW

File: crypto/rand/drbg_hmac.c [hunk 1]
```diff
@@ -203,33 +203,10 @@ int drbg_hmac_init(RAND_DRBG *drbg)
     RAND_DRBG_HMAC *hmac = &drbg->data.hmac;
 
     /*
-     * Confirm digest is allowed. Outside FIPS_MODE we allow all non-legacy
-     * digests. Inside FIPS_MODE we only allow approved digests. Also no XOF
-     * digests (such as SHAKE).
+     * Confirm digest is allowed. We allow all digests that are not XOF
+     * (such as SHAKE).  In FIPS mode, the fetch will fail for non-approved
+     * digests.
```
Comment (mattcaswell) [inline crypto/rand/drbg_hmac.c]: So where do we test for XOF digests?

Comment (paulidale) [reply crypto/rand/drbg_hmac.c]: Oops.  Slipped by.
Commit coming soon.

Comment (mspncp) [reply crypto/rand/drbg_hmac.c]: > Oops. Slipped by.

Me too :-/

**Comment by paulidale** [comment]
Merged to master, thanks all.

**Pull request closed without merging**